test-wasm.html

```diff
@@ -0,0 +1,102 @@
+<script src="wasm/harfbuzzjs.js"></script>
+<script>
+var decoder = new TextDecoder("utf8");
+function bufferToString(buffer) {
+  return decoder.decode(buffer.slice(0, buffer.indexOf(0)));
+}
+
+function hb_tag(tag) {
+  return [...tag].reduce((x, y) => (x << 8) + y.codePointAt(0), 0);
+}
+
+Module().then(module => {
+  window.module = module;
+
+  fetch('test/subset/data/fonts/Roboto-Regular.ttf').then(res => res.arrayBuffer()).then(font => {
+    var buffer = module._malloc(font.byteLength);
+    module.HEAPU8.set(new Uint8Array(font), buffer);
+
+    var blob = module._hb_blob_create(buffer, font.byteLength, 0/*FIXME*/, 0, 0);
+    // module._hb_blob_get_length(blob), module._hb_face_count(blob)
+    var face = module._hb_face_create(blob, 0);
+    var font = module._hb_font_create(face);
+
+    var buffer = module._hb_buffer_create();
+    [...'Your text to shape!'].forEach((x, i) => {
+      module._hb_buffer_add(buffer, x.codePointAt(0), i);
+    });
```

Comment:
How does this handle UTF-16 surrogate pairs (JavaScript used to do funny things with these). Why not use one of `hb_buffer_add_utfXX` functions?

Reply:
Oh I will

Reply:
Well I just wanted to have a working example

Reply:
Gone. [...str] syntax is aware of UTF-16 thing issue.